Update receive page

song50119 opened the pull request:
Resolves #349

song50119 committed:
Update receive page

oshorefueled commented on ui/modal_templates.go:
```diff
@@ -264,6 +265,16 @@ func (m *ModalTemplate) sendInfo() []func(gtx C) D {
+func (m *ModalTemplate) receiveInfo() []func(gtx C) D {
+	return []func(gtx C) D{
+		func(gtx C) D {
+			text := m.th.Body1("Each time you receive a payment, a new address is generated to protect your privacy.")
```
the font size should be 20sp, Body1 is 16sp. The default font weight for labels is 400. The info title should have a font weight of 600.

oshorefueled commented on ui/recieve_page.go:
```diff
@@ -15,194 +23,242 @@ import (
+	newAddrBtn, minInfo, copyBtn decredmaterial.Button
+	infoBtn, moreBtn             decredmaterial.IconButton
+	card                         decredmaterial.Card
+	receiveAddressLabel          decredmaterial.Label
```
try to avoid naming your variable after its type. Your variable name should describe its content not its type. receiveAddress is just fine here.

oshorefueled commented on ui/recieve_page.go:
```diff
@@ -15,194 +23,242 @@ import (
+	pageContainer                layout.List
+	isNewAddr, isInfo            bool
+	addrs                        string
+	newAddrBtn, minInfo, copyBtn decredmaterial.Button
```
try to avoid naming your variable after its type. Your variable name should describe its content not its type.

song50119 committed:
Change field name of receivePage

song50119 committed:
Add pageBackDrop layout

song50119 committed:
Change recieve_page.go to receive_page.go

song50119 committed:
correct text size and button dimensions
- change generate address button TextSize to 16 and padding vertical 20 DP
- remove unuse minInfo button
- change modal cancel button text size to 16
- add SMarginPadding19p5 value to dimensions

oshorefueled commented on ui/values/dimensions.go:
```diff
@@ -53,5 +53,6 @@ var (
 	STextSize20 = unit.Sp(20)
 
-	SMarginPadding16 = unit.Sp(16)
+	SMarginPadding16   = unit.Sp(16)
```
What does the "S" mean in "SMarginPadding16"?

oshorefueled commented on ui/modal_templates.go:
```diff
@@ -371,7 +384,9 @@ func (m *ModalTemplate) Layout(th *decredmaterial.Theme, load *modalLoad) []func
 		func(gtx C) D {
-			return th.H5(load.title).Layout(gtx)
+			t := th.H5(load.title)
+			t.Font.Weight = 600
```
What I meant here was to use "text.Bold" as the font weight and not to assign 600 directly to it

oshorefueled commented on ui/receive_page.go:
```diff
@@ -0,0 +1,594 @@
+	}
+}
+
+func (wg *walletAccountWidget) walletAccountsPopupLayout(gtx layout.Context, name, totalBal, spendableBal string, common pageCommon, wIndex, aIndex int) layout.Dimensions {
```
To many parameters are defined on this method. Try to use very few number of parameters on a method or function to  to keep your code clean and easy to call. I try to limit the number of parameters to 3 or 4 if I can't help it. Anything beyond that, I'll create a data type and pass that to the method. In. You can do the same in this case.

oshorefueled commented on ui/receive_page.go:
```diff
@@ -0,0 +1,594 @@
+type walletAccountWidget struct {
+	title                    decredmaterial.Label
+	walletAccountModal       *decredmaterial.Modal
+	walletsList              layout.List
```
try to avoid using the type as part of the name of a variable or field. In this case, wallets is just fine.

oshorefueled commented on ui/receive_page.go:
```diff
@@ -0,0 +1,594 @@
+
+	return func(gtx C) D {
+		page.Handle(common)
+		page.wallAcctWidget.Handler(common)
```
can this be called from the Handle method for the page?

oshorefueled commented on ui/receive_page.go:
```diff
@@ -0,0 +1,594 @@
+	})
+}
+
+func (pg *receivePage) pageBackDrop(gtx layout.Context) {
```
* backdrop is one word so, "pageBackdrop". 
* Kindly add a comment to this method describing what it does. There's a method named backdrop already, a comment will help to distinguish between the two at a glance.

oshorefueled commented on ui/receive_page.go:
```diff
@@ -0,0 +1,594 @@
+	)
+}
+
+func (pg *receivePage) generateAddressSection(gtx layout.Context, common pageCommon) layout.Dimensions {
```
This seems like it would be better as "titleSection" as the generate address button is hidden until the button is clicked.

oshorefueled commented on ui/receive_page.go:
```diff
@@ -0,0 +1,594 @@
+	)
+}
+
+func (pg *receivePage) addressSection(gtx layout.Context, common pageCommon) layout.Dimensions {
```
This is a row not a section. A section could be demarcated with a separator or space between different card layouts. Other methods that this applies to should be addressed as well.

oshorefueled commented on ui/receive_page.go:
```diff
@@ -0,0 +1,594 @@
+				return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						accountIcon := common.icons.accountIcon
+						accountIcon.Scale = 0.9
```
The exact size of the icon should be used with a scale of 1.0. There seems to be an issue with gio when scaling below 1.0 which causes the image to get clipped on displays with low screen density.

song50119 committed:
Add logo into Qr code

song50119 committed:
Update account selector
- reduce params walletAccountsPopupLayout
- add fields to walletAccount to display on each row of walet accounts moda
- move wallAcctWidget.Handler into page Handler
- pageBackdropLayout layout of background overlay when the popup button generate new address is show,
click outside of the generate new address button to hide the button

oshorefueled commented on ui/receive_page.go:
```diff
@@ -0,0 +1,607 @@
+type walletAccountWidget struct {
+	title                    decredmaterial.Label
+	walletAccount            *decredmaterial.Modal
+	wallets, accounts        layout.List
```
Shouldn't this be defined as a pointer?

song50119 committed:
Display receive address one line

song50119 committed:
Retry generate new address if equal old one

song50119 committed:
Update account selector
- Move on account selector widget to Page common
- Move account selector layouts methods from receive page to component

oshorefueled commented on ui/compontents.go:
```diff
@@ -336,6 +342,212 @@ func endToEndRow(gtx layout.Context, leftWidget, rightWidget func(C) D) layout.D
+	})
+}
+
+func (page *pageCommon) accountTableLayout(gtx layout.Context, leftLabel, rightLabel decredmaterial.Label) layout.Dimensions {
```
This looks like it does the same thing as the "endToEnd" layout. Have a look at the "endToEnd" function in components.go and see how it is used.

oshorefueled commented on ui/compontents.go:
```diff
@@ -336,6 +342,212 @@ func endToEndRow(gtx layout.Context, leftWidget, rightWidget func(C) D) layout.D
+							}),
+							layout.Rigid(func(gtx C) D {
+								spendibleLabel := page.theme.Body2("Spendable")
+								spendibleLabel.Color = page.theme.Color.Gray
```
Kindly correct the spelling here. it should be "spendable", not "spendible". Also try not to add the data type to the variable name, unless it contradicts with its use on the page. For instance, using if "accounts" as a variable name of a List layout would lead to confusion in the code, it would be perfectly fine to make the variable name "accountsList". However, spendable seems just fine here.
